mmv1/third_party/terraform/services/memcache/data_source_memcache_instance_test.go

```diff
@@ -0,0 +1,71 @@
+package memcache_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccMemcacheInstanceDatasourceConfig(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckMemcacheInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMemcacheInstanceDatasourceConfig(context),
+				Check: acctest.CheckDataSourceStateMatchesResourceStateWithIgnores(
+					"data.google_memcache_instance.default",
+					"google_memcache_instance.instance",
+					map[string]struct{}{
+						"resource": {},
+					},
+				),
```

Comment:
Looks like the `ignores` part is unnecessary (there doesn't seem to be a `resource` field on the resource, so the current map isn't doing anything.)

```suggestion
				Check: acctest.CheckDataSourceStateMatchesResourceState(
					"data.google_memcache_instance.default",
					"google_memcache_instance.instance",
				),
```